lib/rake/build.rake

```diff
@@ -111,7 +111,16 @@ namespace :build do
         # The sequencing described here is the best for mitigating any issues
         # that may arise when that best practice is not followed.
         ChatClient.log 'Restarting <b>dashboard</b> Active Job worker(s).'
-        RakeUtils.system 'bin/delayed_job', 'restart'
+        if rack_env?(:production)
+          # WARNING: the number of workers in production is safe to increase,
+          # but is not safe to lower without additional steps. specifically, if
+          # you lower the number of jobs from 10 to 8 (for example), you'll need
+          # to manually kill workers 8 and 9 (zero-based). otherwise, those
+          # workers will continue to run jobs using older code indefinitely.
+          RakeUtils.system 'bin/delayed_job', '-n', '10', 'restart'
```

Comment:
I noticed one other gotcha here, which is that after this PR reaches production, we'll have to manually kill the `delayed_job` process on production-daemon once, because when `-n` is supplied the `restart` command only looks for process names with numerical suffixes:
![Screenshot 2023-10-09 at 3 21 14 PM](https://github.com/code-dot-org/code-dot-org/assets/8001765/2b6f59f4-7fdc-4d9b-9ac9-d11b85d6d154)

I've added a note about this in the PR description.